### PR TITLE
Cleanup of partition related tests and TestPartitionUtils

### PR DESCRIPTION
**Pull request by Donnerbart**
Also added a missing test to `MemberGroupFactoryTest`, to check what
happens if a member doesn't match a `MemberGroupConfig`

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/partition/TestPartitionUtils.java [hunk 1]
```diff
@@ -208,14 +204,9 @@ public void run() {
             latch.countDown();
         }
 
-        private void await() throws InterruptedException {
-            Assert.assertTrue("GetReplicaVersionsRunnable is not executed!", latch.await(1, TimeUnit.MINUTES));
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/partition/TestPartitionUtils.java]: This `latch.await()` was the cause for a ton of `throws InterruptedException` in the test code. I replaced it with `assertOpenEventually()`, which makes it much easier to use the utility method `getDefaultReplicaVersions()`.

File: hazelcast/src/test/java/com/hazelcast/partition/PartitionControlledIdTest.java [hunk 2]
```diff
@@ -225,11 +224,11 @@ public void testList() throws Exception {
     }
 
     @Test
-    public void testSet() throws Exception {
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/partition/PartitionControlledIdTest.java]: Enjoying the new idea feature :)

File: hazelcast/src/test/java/com/hazelcast/partition/SafeClusterTest.java [hunk 3]
```diff
@@ -31,68 +31,68 @@
...
+        boolean safe1 = node1.getPartitionService().isClusterSafe();
+        boolean safe2 = node2.getPartitionService().isClusterSafe();
 
         assertTrue(safe1);
         assertTrue(safe2);
     }
 
     @Test
-    public void isLocalMemberSafe() throws Exception {
-        final HazelcastInstance node = createHazelcastInstance();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/partition/SafeClusterTest.java]: Another final hater :)

Comment (mmedenjak) [reply hazelcast/src/test/java/com/hazelcast/partition/SafeClusterTest.java]: Noooooooooooooooo!!! :)

Comment (Donnerbart) [reply hazelcast/src/test/java/com/hazelcast/partition/SafeClusterTest.java]: I love it for fields, hate it for parameters and mostly dislike it for local variables (methods should be short enough to easily understand them without `final`).

**Pull request merged**